File: android/src/main/java/org/ligi/gobandroid_hd/ui/sgf_listing/item_view_holder/PathViewHolder.java
package org.ligi.gobandroid_hd.ui.sgf_listing.item_view_holder;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;
import butterknife.BindView;
import butterknife.ButterKnife;
import java.io.File;
import org.ligi.gobandroid_hd.R;

public class PathViewHolder extends RecyclerView.ViewHolder implements ViewHolderInterface {

    @BindView(R.id.pathName)
    TextView title;

    public PathViewHolder(View itemView) {
        super(itemView);
        ButterKnife.bind(this, itemView);
    }

    @Override
    public void apply(final File file) {
        title.setText(file.getName());
    }
}

File: android/src/main/java/org/ligi/gobandroid_hd/ui/sgf_listing/item_view_holder/PathViewHolder.kt
package org.ligi.gobandroid_hd.ui.sgf_listing.item_view_holder

import androidx.recyclerview.widget.RecyclerView
import android.view.View
import kotlinx.android.synthetic.main.sgf_dir_list_item.view.*
import java.io.File

class PathViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), ViewHolderInterface {

    override fun apply(file: File) {
        itemView.pathName.text = file.name
    }
}
